component/common/src/main/java/org/exoplatform/commons/cache/future/FutureCache.java

```
@@ -131,7 +131,7 @@ public final V get(final C context, final K key) {
                         retrieval.current = Thread.currentThread();
                         retrieval.future.run();
                     } catch (Exception e) {
-                        log.error("Retrieval of resource " + key + " threw an exception", e);
+                        throw new RuntimeException("Retrieval of resource " + key + " threw an exception", e);
```

Do you think this could impact to every UI place where is using FutureCache in the case of having exception ?

---

An exception should be retrieved on low level API instead of swallowing it. This could cause incoherence in data. In fact, when we swallow exceptions in bottom layer APIs, the transactions on Top layes will commit while there are some operations that failed. On UI, the exceptions are handled anyway, and popups should appears when something wrong happens.